test/e2e/funcs/feature.go

```diff
@@ -133,6 +133,100 @@ func DeploymentBecomesAvailableWithin(d time.Duration, namespace, name string) f
+			return ctx
+		}
+
+		// first wait for pod to be running
+		if err := wait.For(conditions.New(c.Client().Resources()).PodConditionMatch(pod, corev1.PodReady, corev1.ConditionTrue), wait.WithTimeout(d), wait.WithInterval(DefaultPollInterval)); err != nil {
+			t.Errorf("Deployment %s/%s never got a running pod after %s", dp.GetNamespace(), dp.GetName(), since(start))
+			return ctx
+		}
+
+		// now wait to make sure the pod stays running (does not change)
+		start = time.Now()
+		if err := wait.For(conditions.New(c.Client().Resources()).PodConditionMatch(pod, corev1.PodReady, corev1.ConditionFalse), wait.WithTimeout(d), wait.WithInterval(DefaultPollInterval)); err != nil {
+			if deadlineExceed(err) {
+				t.Logf("Deployment %s/%s had running pod that did not change after %s", dp.GetNamespace(), dp.GetName(), since(start))
+			} else {
+				t.Errorf("Error while observing pod for deployment %s/%s", dp.GetNamespace(), dp.GetName())
+			}
+			return ctx
+		}
+		t.Errorf("Deployment %s/%s had pod that changed within %s, but it should not have", dp.GetNamespace(), dp.GetName(), d.String())
+		return ctx
+	}
+}
+
+// ArgSetWithin fails a test if the supplied Deployment does not have a Pod with
+// the given argument set within the supplied duration.
+func ArgSetWithin(d time.Duration, arg, namespace, name string) features.Func {
+	return checkArgSetWithin(d, arg, true, namespace, name)
+}
+
+// ArgUnsetWithin fails a test if the supplied Deployment does not have a Pod with
+// the given argument unset within the supplied duration.
+func ArgUnsetWithin(d time.Duration, arg, namespace, name string) features.Func {
+	return checkArgSetWithin(d, arg, false, namespace, name)
+}
+
+// checkArgSetWithin implements a check for the supplied Deployment having a Pod
+// with the given argument being either set or unset within the supplied
+// duration.
+func checkArgSetWithin(d time.Duration, arg string, wantSet bool, namespace, name string) features.Func {
+	return func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		t.Helper()
+
+		dp := &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Namespace: namespace, Name: name}}
+		t.Logf("Waiting %s for pod in deployment %s/%s to have arg %s set=%t...", d, dp.GetNamespace(), dp.GetName(), arg, wantSet)
+		start := time.Now()
+
+		if err := wait.For(func(ctx context.Context) (done bool, err error) {
+			pod, err := podForDeployment(ctx, t, c, dp)
+			if err != nil {
+				t.Logf("failed to get pod for deployment %s/%s: %s", dp.GetNamespace(), dp.GetName(), err)
+				return false, nil
+			}
+
+			found := false
+			c := pod.Spec.Containers[0]
+			for _, a := range c.Args {
+				if a == arg {
+					found = true
+				}
+			}
+
+			if wantSet && !found {
+				t.Logf("did not find arg %s within %s", arg, c.Args)
+				return false, nil
+			} else if !wantSet && found {
+				t.Logf("unexpectedly found arg %s within %s", arg, c.Args)
+				return false, nil
+			}
+
+			return true, nil
```

Comment:
Style nit: I think this all could be a switch statement.

Reply:
made this style change to a `switch` statement in https://github.com/crossplane/crossplane/pull/6182/commits/93f96e6eff93f256333256a0c80b50417fcafadb